sql/parser/type_check_test.go

```diff
@@ -26,6 +26,12 @@ import (
 
 func TestTypeCheckExpr(t *testing.T) {
 	testData := []string{
+		`NULL + 1`,
+		`NULL + 1.1`,
+		`NULL + '2006-09-23'::date`,
+		`NULL + '1h'::interval`,
+		`NULL + 'hello'`,
+		`NULL + 'hello'::bytes`,
```

Comment:
Perhaps one of these should be `<constant> + NULL` (i.e. `NULL` on the right hand side).